File: 2022/04/a.py
def contains(haystack, needle):
    if haystack[0] <= needle[0] and haystack[1] >= needle[1]:
        return 1
    return 0

def part_two(haystack, needle):
    if needle[0] >= haystack[0] and needle[0] <= haystack[1]:
        return 1
    elif needle[1] >= haystack[0] and needle[1] <= haystack[1]:
        return 1
    return 0

file = open("input.txt")
lines = file.readlines()

ret = 0
for line in lines:
    split = line.split(',')
    first = list(map(int, split[0].split('-')))
    second = list(map(int, split[1].split('-')))
    if first[1] - first[0] < second[1] - second[0]:
        ret += part_two(second, first)
    else:
        ret += part_two(first, second)
print(ret)

File: 2022/04/a.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::path::Path;

struct Section {
    start: u16,
    end: u16,
}

fn get_number(s: String, first: bool) -> Option<u16>
{
    if first {
        s.find('-').map(|i| s[..i].parse().ok()).flatten()
    }
    else {
        s.find('-').map(|i| s[i+1..].parse().ok()).flatten()
    }
}

fn main() {
    let mut contains = 0;
    if let Ok(lines) = read_lines("./input.txt") {
        for line in lines {
            if let Ok(sections) = line {
                let v: Vec<&str> = sections.split(",").collect();
                let first = Section {
                    start: get_number(v[0].to_string(), true),
                    end: get_number(v[0].to_string(), false),
                };
                let second = v[1].to_string();
                
                println!("{}", first);
            }
        }
    }
}

fn read_lines<P>(filename: P) -> io::Result<io::Lines<io::BufReader<File>>>
where P: AsRef<Path>, {
    let file = File::open(filename)?;
    Ok(io::BufReader::new(file).lines())
}